src/rules/no-self-import.js

```diff
@@ -0,0 +1,42 @@
+/**
+ * @fileOverview Prohibits a file from importing itself
+ * @author Gio d'Amelio
+ */
+
+import resolve from 'eslint-module-utils/resolve'
+import isStaticRequire from '../core/staticRequire'
+
+function isImportingSelf(context, node, requireName) {
+  const filePath = context.getFilename()
+  const resolvedPath = resolve(requireName, context)
+  if (filePath === resolvedPath) {
+    context.report({
+        node,
+        // TODO: check in about message, should it print the name of the module itself
```

Comment:
yes, i think it should - it's much more helpful that way. Ideally, it'd include the path to the file, and, *if different*, the path used in the `import` statement.